src/internal.c

```diff
@@ -12062,7 +12062,7 @@ int TimingPadVerify(WOLFSSL* ssl, const byte* input, int padLen, int macSz,
     good |= MaskMac(input, pLen, ssl->specs.hash_size, verify);
 
     /* Non-zero on failure. */
-    good = ~good;
+    good = ~(word32)good;
```

Comment:
These are the same type, why is this needed?

Reply:
Can you run these casts by Sean?

Reply:
Sure, I will run these by Sean.

Based on my research, the "~" operator first promotes "good" from a byte (unsigned char) to a signed integer here before doing the operation.  Since semantics for a signed int can vary, casting to an unsigned integer guarantees the value used is the unsigned binary as expected.

Reply:
Are you positive that none of our supported compilers will complain about assigning 32 bits to 8 bits?

Reply:
Not completely, just for the compilers I have tested so far:

- Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn) - OSX 10.9.5
- gcc 7.2.0 - Ubuntu 17.10
- clang 4.0.1-6 - Ubuntu 17.10
- Visual Studio 2015 Community - Windows 10
- IAR-EWARM 6.40.5.4275 - Windows 7 Pro

One option would be to cast the result back to the original type, which should suppress any 32 to 8-bit or similar warnings:

```
good = (byte)~(word32)good;
```

The other option would be to suppress the [Pa091] warning when building for IAR-EWARM projects.